src/main/java/seedu/agendum/model/task/UniqueTaskList.java

```diff
@@ -73,6 +74,86 @@ public boolean remove(ReadOnlyTask toRemove) throws TaskNotFoundException {
         }
         return taskFoundAndDeleted;
     }
+    
+    /**
+     * Renames the equivalent task in the list.
+     *
+     * @throws TaskNotFoundException if no such task could be found in the list.
+     * @throws DuplicateTaskException if the renamed task is a duplicate of an existing task in the list.
+     */
+    public boolean rename(ReadOnlyTask toRename, Name newTaskName)
+            throws TaskNotFoundException, DuplicateTaskException {
+        assert toRename != null;
+        assert newTaskName != null;
+        final int taskIndex = internalList.indexOf(toRename);
+        final boolean taskFoundAndRenamed = (taskIndex != -1);
+        if (!taskFoundAndRenamed) {
+            throw new TaskNotFoundException();
+        }
+        Task toCheck = new Task(toRename);
+        toCheck.setName(newTaskName);
+        if (contains(toCheck)) {
+            throw new DuplicateTaskException();
+        }
+        internalList.set(taskIndex, toCheck);
+        return taskFoundAndRenamed;
+    }
+
+    /**
+     * Schedules the equivalent task in the list.
+     *
+     * @throws TaskNotFoundException if no such task could be found in the list.
+    */
+    public boolean schedule(ReadOnlyTask toSchedule, LocalDateTime startDateTime,
+            LocalDateTime endDateTime) throws TaskNotFoundException {
+        assert toSchedule != null;
+        final int taskIndex = internalList.indexOf(toSchedule);
+        final boolean taskFoundAndScheduled = (taskIndex != -1);
+        if (!taskFoundAndScheduled) {
+            throw new TaskNotFoundException();
+        }
+        Task scheduledTask = new Task(toSchedule);
+        scheduledTask.setStartDateTime(startDateTime);
+        scheduledTask.setEndDateTime(endDateTime);
+        internalList.set(taskIndex, scheduledTask);
+        return taskFoundAndScheduled;
+    }
+
+    /**
+     * Marks the equivalent task in the list.
+     *
+     * @throws TaskNotFoundException if no such task could be found in the list.
+     */
+    public boolean mark(ReadOnlyTask toMark) throws TaskNotFoundException {
+        assert toMark != null;
+        final int taskIndex = internalList.indexOf(toMark);
+        final boolean taskFoundAndMarked = (taskIndex != -1);
+        if (!taskFoundAndMarked) {
+            throw new TaskNotFoundException();
+        }
+        Task markedTask = new Task(toMark);
+        markedTask.markAsCompleted();
+        internalList.set(taskIndex, markedTask);
```

Comment:
The task is replaced by a clone. For rename, it will automatically be updated in the UI but not for mark/unmark yet